src/System.Private.CoreLib/shared/System/Reflection/Assembly.cs

```diff
@@ -120,7 +120,7 @@ public virtual Type[] GetTypes()
         public virtual event ModuleResolveEventHandler ModuleResolve { add { throw NotImplemented.ByDesign; } remove { throw NotImplemented.ByDesign; } }
 
         public virtual Module? ManifestModule { get { throw NotImplemented.ByDesign; } }
```

Comment:
`ManifestModule` property should be non-null.

Reply:
I guess this was marked as nullable because of this: https://github.com/dotnet/coreclr/blob/abdee1f2fa9531c78eeadef6d0bb8fb1f823d011/src/vm/runtimehandles.cpp#L2758

In RuntimeAssembly we even check for it not being null:
https://github.com/dotnet/coreclr/blob/1ab8b90ca9f8748776b00cc21cd5740b53dec67b/src/System.Private.CoreLib/src/System/Reflection/RuntimeAssembly.cs#L106-L115

Should we still make it non-null?

Reply:
I addressed this feedback but I'm still hesitant about making ManifestModule return non-null

Reply:
This is some sort of defensive programming that does not make sense. You can delete both of these null checks.

I am pretty sure that `ManifestModule` should be non-null.

Reply:
Ok, cool. I'll remove those checks then.